skimage/feature/texture.py

```diff
@@ -97,11 +99,21 @@ def greycomatrix(image, distances, angles, levels=256, symmetric=False,
     assert_nD(distances, 1, 'distances')
     assert_nD(angles, 1, 'angles')
 
-    assert levels <= 256
     image = np.ascontiguousarray(image)
     assert image.min() >= 0
-    assert image.max() < levels
-    image = image.astype(np.uint8)
+
+    image_max = image.max()
```

Comment:
> The default alternative has to be the same as the current one. Otherwise, it will be an API change and we have to follow deprecation path.
I get your point. 

> Also, the might be many zeros within the [0:levels] range. As for me, clipping from the upper bound of a range doesn't seem to be the right simplification.
You are right in principle, but in practice there are many cases where the full range is not used (i.e. most levels are used in [0, max], but max << max(datatype)). 

> By the way, for uint16, if have set all things correctly, there will be 65536**2 * 2 bytes matrix, what equals to ~8GB. Not suitable.
I know. This was the reason why I changed the default, so that the function can still support some 16bit images (those that do not use a close a large data range). 

 As I mentioned before, my concrete application was to use the function on label images in order to retrieve all touching regions. There are two options: I write a dedicated function to retrieve the touching regions (I do not know whether skimage would be interested in this). Or I give back a sparse matrix. This should solve the problem for all reasonably sized 16 bit images.